jgrapht-core/src/main/java/org/jgrapht/intervalgraph/CenteredIntervalTree.java

```diff
@@ -0,0 +1,110 @@
+package org.jgrapht.intervalgraph;
+
+import org.jgrapht.Graph;
+import org.jgrapht.intervalgraph.interval.Interval;
+import org.omg.SendingContext.RunTime;
+
+import java.util.*;
+
+public class CenteredIntervalTree<T extends Comparable<T>> implements IntervalGraph<T>{
+    private T centerPoint;
+    private CenteredIntervalTree<T> leftTree; // contains all the intervals *completely* to the left of the center point
+    private CenteredIntervalTree<T> rightTree; // contains all the intervals *completely* to the right of the center point
+    private List<Interval<T>> intersectionsByStart = new LinkedList<>(); // intervals intersecting center point, sorted by start point
+    private List<Interval<T>> intersectionsByEnd = new LinkedList<>(); // ..., sorted by end point
+
+    public CenteredIntervalTree(T centerPoint, List<Interval<T>> intervals) {
+        this.centerPoint = centerPoint;
+        compute(intervals);
+    }
+
+    public CenteredIntervalTree(List<Interval<T>> intervals) {
+        this.centerPoint = intervals.get(0).getEnd(); // TODO replace this with something, its arbitrary
+        compute(intervals);
```

Comment:
This might be personal taste, but usually people expect that the constructor runs in constant time. I would change it to be private and instead offer a static method to get an instance of this class.